packages/laravel/src/Concerns/HasSharedProperties.php

```diff
@@ -25,6 +25,18 @@ public function share(string|array|Arrayable $key, mixed $value = null): static
         return $this;
     }
 
+    /**
+     * Shares data to every response if the condition is true.
+     */
+    public function shareWhen(bool|\Closure $condition, string|array|Arrayable $key, mixed $value = null): static
+    {
+        if ($condition === true || ($condition instanceof \Closure && $condition() === true)) {
```

Comment:
```suggestion
        if ($condition instanceof \Closure ? $condition($this) : $condition) {
```

No need to check for `true`, truthy/falsy is enough.